Mathlib/FieldTheory/RatFunc/AsPolynomial.lean

```diff
@@ -207,6 +210,61 @@ theorem valuation_of_mk (f : Polynomial K) {g : Polynomial K} (hg : g ≠ 0) :
       (Polynomial.idealX K).intValuation f / (Polynomial.idealX K).intValuation g := by
   simp only [RatFunc.mk_eq_mk' _ hg, valuation_of_mk']
 
+end HeightOneSpectrum
+
+section TrivialOnConstants
+
+variable {Γ : Type*} [LinearOrderedCommMonoidWithZero Γ] {v : Valuation (RatFunc K) Γ}
+
+open Valuation
+
+/--
+If a valuation is trivial on constants then the valuation of `(monomial n a)` is equal
+to `(v (RatFunc.X)) ^ n`. -/
+lemma valuation_monomial_eq_valuation_X_pow
+    (hv : ∀ a : K, a ≠ 0 → v (algebraMap K (RatFunc K) a) = 1) (n : ℕ) {a : K} (ha : a ≠ 0) :
+    v (monomial n a) = v RatFunc.X ^ n := by
+  have h : algebraMap _ (RatFunc K) (C a) = algebraMap _ _ a := rfl
+  simp only [RatFunc.coePolynomial, ← C_mul_X_pow_eq_monomial, _root_.map_mul, _root_.map_pow,
+    RatFunc.algebraMap_X, h, hv a ha, one_mul]
+
+/--
+If a valuation is trivial on constants and `1 < v RatFunc.X` then
+`v p = v RatFunc.X ^ p.natDegree`. -/
+theorem valuation_eq_valuation_X_pow_natDegree_of_one_lt_valuation_X
+    [Nontrivial Γ] [PosMulStrictMono Γ] (hlt : 1 < v RatFunc.X)
```

Comment:
`hlt` seems a strange assumption, the "usual"  valuation of `X` is `< 1`...

Reply:
Ohhh, I see! I'd probably add a line of doc saying that "the condition `hlt` is typically satisfied by the valuation at infinity" (or something like this), but beyond that you can ignore this comment.